pkg/controller/pkg/revision/establisher.go

```diff
@@ -131,30 +133,42 @@ func (e *APIEstablisher) Establish(ctx context.Context, objs []runtime.Object, p
 	return resourceRefs, nil
 }
 
-func (e *APIEstablisher) create(ctx context.Context, obj resource.Object, parent resource.Object, control bool, opts ...client.CreateOption) error {
+func (e *APIEstablisher) create(ctx context.Context, obj runtime.Object, parent resource.Object, control bool, opts ...client.CreateOption) error {
+	o := obj.DeepCopyObject().(resource.Object)
 	ref := meta.AsController(meta.TypedReferenceTo(parent, parent.GetObjectKind().GroupVersionKind()))
 	if !control {
 		ref = meta.AsOwner(meta.TypedReferenceTo(parent, parent.GetObjectKind().GroupVersionKind()))
 	}
 	// Overwrite any owner references on the desired object.
-	obj.SetOwnerReferences([]metav1.OwnerReference{ref})
+	o.SetOwnerReferences([]metav1.OwnerReference{ref})
```

Comment:
Won't this be a no-op? It looks like we create a deep copy of obj, then change its owner references, then apply obj. Do we need to make the deepcopy, or can we just assert obj to a `resource.Object` (or `metav1.Object`)?

Reply:
> Won't this be a no-op?

This is to guard against a case where a package attempts to install a resource with owner references already present. The reason why I have moved to deep copying in both the create and update methods is that we call them multiple times (once for dry run, once for actual action), and when we modify the owner references on the first pass, they may be inaccurate on the second (e.g. I noticed pointer values being modified, which can be unexpected if you have already stored the object into a different struct, such as `allObjs` here).

Reply:
I'm still not quite following here; are you intending to be applying the copy (`o`) below, or the original (`obj`)?

Reply:
Ah yes sorry, I was looking at the comment view here, that omits that line.. will fix